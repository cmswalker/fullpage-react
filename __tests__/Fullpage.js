import React from 'react';
import { Fullpage, Slide, HorizontalSlider } from '../FullpageReact';
import renderer from 'react-test-renderer';

const fullPageOptions = {
	scrollSensitivity: 7,
	touchSensitivity: -3,
	scrollSpeed: 500,
	resetSlides: true,
	hideScrollBars: true,
	enableArrowKeys: true
};

const createNodeMock = () => {
  return {
    addEventListener: () => {}
  };
};

const killWindow = () => {
  global.__window__ = global.window;
  global.window = undefined;
  window = global.window;
};

const restoreWindow = (copy) => {
  global.window = global.__window__;
  window = global.window;
}

describe('FullpageReact', () => {

  describe('with Slides', () => {

    beforeEach(() => {
      const verticalSlides = [
        <Slide style={{backgroundColor: 'blue'}}><p>1</p></Slide>,
        <Slide style={{backgroundColor: 'pink'}}><p>2</p></Slide>
      ];
      fullPageOptions.slides = verticalSlides;
    });

    it('should support vertical slides', () => {
      const component = renderer.create(
        <Fullpage {...fullPageOptions}/>,
        {createNodeMock}
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('with HorizontalSlides', () => {

    beforeEach(() => {
      const horizontalSliderProps = {};
      const horizontalSlides = [
        <Slide style={{backgroundColor: 'red'}}><p>Horizontal 1</p></Slide>,
        <Slide style={{backgroundColor: 'yellow'}}><p>Horizontal 2</p></Slide>,
        <Slide style={{backgroundColor: 'green'}}><p>Horizontal 3</p></Slide>
      ];
      horizontalSliderProps.slides = horizontalSlides;

      const verticalSlides = [
        <Slide style={{backgroundColor: 'blue'}}><p>1</p></Slide>,
        <HorizontalSlider {...horizontalSliderProps} infinite={true} name={'horizontal1'}/>
      ];
      fullPageOptions.slides = verticalSlides;
    });

    it('should support horizontal slides', () => {
      const component = renderer.create(
        <Fullpage {...fullPageOptions}/>,
        {createNodeMock}
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('with SSR', () => {
    beforeEach(() => {
      const verticalSlides = [<Slide style={{ backgroundColor: 'blue' }}>
          <p>1</p>
        </Slide>, <Slide style={{ backgroundColor: 'pink' }}>
          <p>2</p>
        </Slide>];
      fullPageOptions.slides = verticalSlides;

      killWindow();
    });

    afterEach(() => {
      restoreWindow();
    })

    it('should mount via SSR', () => {
      const component = renderer.create(
        <Fullpage {...fullPageOptions} />,
        { createNodeMock }
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should unmount via SSR', () => {
      const component = renderer.create(
        <Fullpage {...fullPageOptions} />,
        { createNodeMock }
      );

      component.unmount();
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    })
  });

});
