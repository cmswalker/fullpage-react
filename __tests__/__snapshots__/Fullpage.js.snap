// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullpageReact with HorizontalSlides should support horizontal slides 1`] = `
<div
  className="Fullpage"
>
  <div
    className="VerticalSlide"
    data-horizontal-slider={null}
    data-slide="VerticalSlide"
    id={undefined}
    style={
      Object {
        "backgroundColor": "blue",
        "height": "0px",
        "overflow": "hidden",
        "position": "relative",
        "width": "0px",
      }
    }
  >
    <p>
      1
    </p>
  </div>
  <div
    className="HorizontalSlider"
    data-name="horizontal1"
    data-slide="HorizontalSlider"
    id={undefined}
    style={
      Object {
        "height": "0px",
        "margin": "0",
        "overflowX": "hidden",
        "padding": "0px",
        "position": "relative",
        "whiteSpace": "nowrap",
        "width": "0px",
      }
    }
  >
    <div
      className="HorizontalSlide"
      data-horizontal-slider="horizontal1"
      data-slide="HorizontalSlide"
      id={undefined}
      style={
        Object {
          "backgroundColor": "red",
          "height": "0px",
          "overflow": "hidden",
          "position": "relative",
          "width": "0px",
        }
      }
    >
      <p>
        Horizontal 1
      </p>
    </div>
    <div
      className="HorizontalSlide"
      data-horizontal-slider="horizontal1"
      data-slide="HorizontalSlide"
      id={undefined}
      style={
        Object {
          "backgroundColor": "yellow",
          "height": "0px",
          "overflow": "hidden",
          "position": "relative",
          "width": "0px",
        }
      }
    >
      <p>
        Horizontal 2
      </p>
    </div>
    <div
      className="HorizontalSlide"
      data-horizontal-slider="horizontal1"
      data-slide="HorizontalSlide"
      id={undefined}
      style={
        Object {
          "backgroundColor": "green",
          "height": "0px",
          "overflow": "hidden",
          "position": "relative",
          "width": "0px",
        }
      }
    >
      <p>
        Horizontal 3
      </p>
    </div>
  </div>
</div>
`;

exports[`FullpageReact with SSR should mount via SSR 1`] = `
<div
  className="Fullpage"
>
  <div
    className="VerticalSlide"
    data-horizontal-slider={null}
    data-slide="VerticalSlide"
    id={undefined}
    style={
      Object {
        "backgroundColor": "blue",
        "height": "0px",
        "overflow": "hidden",
        "position": "relative",
        "width": "0px",
      }
    }
  >
    <p>
      1
    </p>
  </div>
  <div
    className="VerticalSlide"
    data-horizontal-slider={null}
    data-slide="VerticalSlide"
    id={undefined}
    style={
      Object {
        "backgroundColor": "pink",
        "height": "0px",
        "overflow": "hidden",
        "position": "relative",
        "width": "0px",
      }
    }
  >
    <p>
      2
    </p>
  </div>
</div>
`;

exports[`FullpageReact with SSR should unmount via SSR 1`] = `null`;

exports[`FullpageReact with Slides should support vertical slides 1`] = `
<div
  className="Fullpage"
>
  <div
    className="VerticalSlide"
    data-horizontal-slider={null}
    data-slide="VerticalSlide"
    id={undefined}
    style={
      Object {
        "backgroundColor": "blue",
        "height": "0px",
        "overflow": "hidden",
        "position": "relative",
        "width": "0px",
      }
    }
  >
    <p>
      1
    </p>
  </div>
  <div
    className="VerticalSlide"
    data-horizontal-slider={null}
    data-slide="VerticalSlide"
    id={undefined}
    style={
      Object {
        "backgroundColor": "pink",
        "height": "0px",
        "overflow": "hidden",
        "position": "relative",
        "width": "0px",
      }
    }
  >
    <p>
      2
    </p>
  </div>
</div>
`;
